final_task/rss_reader/converter.py

```diff
@@ -0,0 +1,153 @@
+    else:
+        fontpath = os.path.join(this_directory, font)
+
+
+def convert_pdf(rss_news_clean: dict, path: str):
+    """This function creates pdf"""
+
+    pdf = FPDF()
+    pdf.add_font('arial_uni', '', fontpath, True)
+    pdf.set_margins(10, 10, 10)
+    counter = 1
+
+    for rss_news in rss_news_clean.values():
+        pdf.add_page(('P', 'A4'))
+
+        if rss_news['image']:
+            try:
+                url.urlretrieve(rss_news['image'], str(counter)+'imageTemp.jpg')
+                pdf.image(str(counter)+'imageTemp.jpg', 120, 12, 50, 50, link=rss_news['image'])
+                os.remove(str(counter)+'imageTemp.jpg')
+                counter += 1
+            except RuntimeError:
+                pdf.set_line_width(1)
+                pdf.rect(120, 12, 50, 50)
+                pdf.set_xy(120, 12)
+                pdf.set_font("Arial", "B", size=14)
+                pdf.multi_cell(w=50, h=12, align='C', txt=' \n IMAGE IS CORRUPTED')
+            except url.HTTPError:
+                pdf.set_line_width(1)
+                pdf.rect(120, 12, 50, 50)
+                pdf.set_xy(120, 12)
+                pdf.set_font("Arial", "B", size=14)
+                pdf.multi_cell(w=50, h=12, align='C', txt=' \n IMAGE IS CORRUPTED')
+            except url.URLError:
+                pdf.set_line_width(1)
+                pdf.rect(120, 12, 50, 50)
+                pdf.set_xy(120, 12)
+                pdf.set_font("Arial", "B", size=14)
+                pdf.multi_cell(w=50, h=12, align='C', txt=' \n NO INTERNET CONNECTION')
+            except ValueError:
+                pdf.set_line_width(1)
+                pdf.rect(120, 12, 50, 50)
+                pdf.set_xy(120, 12)
+                pdf.set_font("Arial", "B", size=14)
+                pdf.multi_cell(w=50, h=12, align='C', txt=' \n NO PICTURE AVAILABLE')
+
+        else:
+            pdf.set_line_width(1)
+            pdf.rect(120, 12, 50, 50)
+            pdf.set_xy(120, 12)
+            pdf.set_font("Arial", "B", size=14)
+            pdf.multi_cell(w=50, h=12, align='C', txt=' \n NO PICTURE AVAILABLE')
+
+        pdf.set_font("arial_uni", size=12)
+        string = 'Feed: ' + rss_news['feed'] + '\n' + '   ' + '\n' 'Title: ' + rss_news['title'] + '\n' + '   ' + '\n'
+        string += 'Date: ' + rss_news['date'] + '\n' + '   ' + '\n'
+        pdf.multi_cell(w=100, h=6, txt=string)
+
+        pdf.set_font("arial_uni", size=12)
+        string = 'News: ' + '\n' + '   \n' + str(rss_news['description'].encode('utf-8', 'ignore').decode('utf-8'))
+        pdf.write(h=6, txt=string)
+
+        pdf.set_font("Arial", "B", size=12)
+        pdf.write(h=6, txt= 'Link: ' + rss_news['link'],link=rss_news['link'])
+
+    pdf.output(path + 'news.pdf', 'F')
+
+
+def convert_log_pdf(log_journal: dict, path: str):
+    """This function converts log journal to pdf"""
+
+    pdf = FPDF()
+    pdf.set_font('Arial', 'B', size=14)
+    pdf.set_margins(10, 10, 10)
+    pdf.add_page(('P', 'A4'))
```

Comment:
You had the same chunk of code above, it is better to move it to a separate function

Reply:
The code above is not the same as above. The main difference is the font: arial_uni for the first case, Arial for the second case.